layers/meta-tegra-support/recipes-bsp/u-boot/u-boot-tegra_%.bbappend

```diff
@@ -0,0 +1,6 @@
+# Work around Mender's installation of this file (symlink)
+# in its libubootenv bbappend.
+do_install_append_mender-uboot() {
+    rm -f ${D}${sysconfdir}/fw_env.config
+}
+
```

Comment:
Not too long ago, oecore started adding an `RRECOMMENDS` for `u-boot-default-env`, which started exposing this conflict. https://github.com/mendersoftware/meta-mender/commit/d25feefe9912f31373067097f891d8b5c538bded adds an `RPROVIDES` to libubootenv, so it can be the provider of that file instead, and that change was a enough to fix the conflict for me in the normal build.

What you've done here is necessary for the SDK build. I thought the following change to the libubootenv append might work, but it didn't:

```
RPROVIDES_${PN}_class-nativesdk += "u-boot-default-env"
```

Is there a way we can fix this that we could upstream?

Reply:
Yeah, it was SDK builds that were triggering the issue for me.

The right way to fix this is for meta-mender-core *not* to use libubootenv to supply this file.  My understanding is that libubootenv is not supposed to be machine-specific; it depends on the u-boot recipe to package up and supply the needed machine-specific files - not just fw_env.config, but also the default environment settings that get compiled into U-Boot... I think that's what led to the RRECOMMENDS addition.

I haven't dug into this enough to understand why Mender is doing it this way.  If for some reason the config/default env file can't just be provided by the u-boot build, having a separate MACHINE-specific recipe to do so which RPROVIDES/RREPLACES/RCONFLICTS with `u-boot-default-env` should be enough, I think.

Reply:
OK. This makes enough sense for now.